plugin-client/src/main/java/com/cloudera/nav/plugin/client/NavApiCient.java

```diff
@@ -71,18 +78,105 @@ public void registerModels(Collection<Object> models) {
    * @return a collection of available sources
    */
   public Collection<Source> getAllSources() {
+    System.out.println("here");
     RestTemplate restTemplate = new RestTemplate();
-    String url = getSourceUrl();
+    String url = getUrl();
     HttpHeaders headers = getAuthHeaders();
+    System.out.println(headers);
     HttpEntity<String> request = new HttpEntity<String>(headers);
+    Class<SourceAttrs[]> sourceAttrsClass = SourceAttrs[].class;
     ResponseEntity<SourceAttrs[]> response = restTemplate.exchange(url,
-        HttpMethod.GET, request, SourceAttrs[].class);
+        HttpMethod.GET, request, sourceAttrsClass);
+//    ResponseEntity<SourceAttrs[]> response = restTemplate.getForEntity(url, )
+    System.out.println(response);
     Collection<Source> sources = Lists.newArrayList();
     for (SourceAttrs info : response.getBody()) {
       sources.add(info.createSource());
     }
     return sources;
   }
+  /** Returns all of the entities and relations in the database, plus a marker to denote when this search took place
+   *
+   * @return
+   */
+  public UpdatedResults getAllUpdated(){
+    UpdatedResults updatedResults;
+    Map<String, Integer> currentMarker = getCurrentMarker();
+    try {
+      String currentMarkerRep = new ObjectMapper().writeValueAsString(currentMarker);
+      String queryString = "";
+      updatedResults = aggUpdatedResults(currentMarkerRep, queryString);
+      return updatedResults;
+    } catch (IOException e){
+      System.err.println(e.getMessage());
+      throw Throwables.propagate(e);
+    }
+  }
+
+  /**Returns all of the entities and relations in the database that have been updated or added since the source iterations indicated by the marker
+   *
+   * @param markerRep JSON representation of sourceId : sourcesourceExtractIteration
```

Comment:
the fact that the markerRep is JSON representation of the extractorRunId map is an implementation detail and should not appear in the javadoc.